scripts/install/install_unix.sh

```diff
@@ -13,165 +13,638 @@
+  fi
+
+  if [ -z "$version" ] ; then
+    collector_download_url="$url/latest/download/${PACKAGE_NAME}_${os_arch}.${package_type}"
+  else
+    collector_download_url="$url/download/v$version/${PACKAGE_NAME}_${os_arch}.${package_type}"
+  fi
+}
+
+# This will check all prerequisites before running an installation.
+check_prereqs()
+{
+  banner "Checking Prerequisites"
+  increase_indent
+  root_check
+  os_check
+  os_arch_check
+  package_type_check
+  dependencies_check
+  success "Prerequisite check complete!"
+  decrease_indent
+}
+
+# This
+root_check()
+{
+  system_user_name=$(id -un)
+  if [ "${system_user_name}" != 'root' ]
+  then
+    failed
+    error_exit "Script needs to be run as root or with sudo permissions"
+  fi
+}
+
+# This will check if the operating system is supported.
+os_check()
+{
+  info "Checking that the operating system is supported..."
+  os_type=$(uname -s)
+  case "$os_type" in
+    Linux)
+      succeeded
+      ;;
+    *)
+      failed
+      error_exit "The operating system $(fg_yellow "$os_type") is not supported by this script."
+      ;;
+  esac
+}
+
+# This will check if the system architecture is supported.
+os_arch_check()
+{
+  info "Checking for valid operating system architecture..."
+  arch=$(uname -m)
+  case "$arch" in 
+    x86_64|aarch64|arm64|aarch64_be|armv8b|armv8l|arm|armv6l|armv7l)
+      succeeded
+      ;;
+    *)
+      failed
+      error_exit "The operating system architecture $(fg_yellow "$arch") is not supported by this script."
+      ;;
+  esac
 }
 
-set_package_type() {
-    if command -v dpkg > /dev/null 2>&1; then
-        package_type="deb"
-    elif command -v rpm > /dev/null 2>&1; then
-        package_type="rpm"
+# This will check if the current environment has
+# all required shell dependencies to run the installation.
+dependencies_check()
+{
+  info "Checking for script dependencies..."
+  FAILED_PREREQS=''
+  for prerequisite in $PREREQS; do
+    if command -v "$prerequisite" >/dev/null; then
+      continue
     else
-        echo "Installation failed: Could not find dpkg or rpm on the system"
-        exit 1
+      if [ -z "$FAILED_PREREQS" ]; then
+        FAILED_PREREQS="${fg_red}$prerequisite${reset}"
+      else
+        FAILED_PREREQS="$FAILED_PREREQS, ${fg_red}$prerequisite${reset}"
+      fi
     fi
+  done
+
+  if [ -n "$FAILED_PREREQS" ]; then
+    failed
+    error_exit "The following dependencies are required by this script: [$FAILED_PREREQS]"
+  fi
+  succeeded
+  return 0
 }
 
-set_file_name() {
-    package_file_name="observiq-otel-collector_linux_${arch}.${package_type}"
-    out_file_path="$TMP_DIR/$package_file_name"
+# This will check to ensure either dpkg or rpm is installedon the system
+package_type_check()
+{
+  info "Checking for package manager..."
+  if command -v dpkg > /dev/null 2>&1; then
+      succeeded
+      pkg_type="deb"
+      return 0
+  elif command -v rpm > /dev/null 2>&1; then
+      succeeded
+      pkg_type="rpm"
+      return 0
```

Comment:
I'm noticing a lot of inconsistency with how we're handling errors.

In general, should the called functions be returning a status code (`return 0` on success, `return 1` on failure) and let the caller handle exiting and printing an error, or should the called function handle erroring. 

IMO, in general, we should just let the function handle erroring since it has finer granularity with what actually went wrong.